EXTENSIONS.md

```diff
@@ -0,0 +1,71 @@
+
+## Open-source "third-party" extensions
+We refer to all other extensions not implemented nor officially maintained by the Gen AI team as open-source third-party extensions.
+
+### Ownership and location
+Open-source third-party extensions can live anywhere, inside [IBM organization](https://github.com/IBM/), the wider Github universe, or other contributors' own Github spaces (e.g., other IBM teams, clients, partners). Maintaining these extensions would be the responsibility of their owners.
+
+
+
+# Extensions migration
+Extensions might be migrated from one type to another upon evaluation and approval of the Gen AI maintenance team. Below, we describe three common migration scenarios:
+
+## Open-source "third-party" to open-source "Gen AI official"
+If open-source third-party contributors have an interest in their extensions becoming official, they must contact the IBM Gen AI team. The Gen AI team will evaluate the request considering novelty and implementation quality (meaning it was developed following the Gen AI official guide and meets Gen AI extensions' design considerations). If the request is approved, the Gen AI team will assist the extension owners in migrating the extension.
+
+## Open-source "Gen AI official" to Gen AI's core
+Upon evaluation of the Gen AI maintenance team, open-source extensions could be merged with Gen AI core code and become deprecated as extensions. Deprecation announcements will be shared as part of the release notes of every new version of the IBM Gen AI package.
+
+# Design considerations to build extensions
+To design an extension for IBM Gen AI SDK a developer should consider a mix-and-match of the following two important development experience patterns:
+
+## Extensions that simulate being part of a core class
+
+In spite of the extension's code living in a separate location, **try to give the impression that extended functionality is part of an IBM Gen AI core class**.
+
+The code snippet below shows an example of this pattern. While `from_template` method is part of an extension class (`PromptExtension`) and not a core class (`PromptPattern`), the way it is being called gives end-users the impression the method is part of `PromptPattern` core class. The magic that enables providing this development experience is the use of decorators at import time. This pattern enables extending user-facing classes in a user-friendly way.
+
+`genai_prompt_pattern = PromptPattern.langchain.from_template(langchain_prompt_template)  # notice langchain hook in the function call.`
+
+## Extensions that wrap core class functionality
+
+Designing extensions that do not integrate, but  **wrap functionality in Gen AI's core** for common use cases. These classes are meant to be used as entities on their own.
+
+The `LangChainInterface` class (see below) is an example of this pattern. `LangChainInterface` is a subclass of langchain's `LLM` class that wraps `generate` functionality of Gen AI's Model.
```

Comment:
Maybe end this description something like "For full definition of this class, see <link-from-langchain-file-main-branch>"?